pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/PanDomainAuthSettingsRefresher.scala

```diff
@@ -19,20 +19,22 @@ import scala.concurrent.duration.FiniteDuration
   *
   * @param domain the domain you are authin agains
   * @param system the identifier for your app, typically the same as the subdomain your app runs on
+  * @param bucketName the bucket where the settings are stored
   * @param actorSystem the actor system to create the refresh actor
-  * @param awsCredentialsProvider optional credential provider
-  * @param awsRegion optional region
+  * @param awsCredentialsProvider AWS credential provider
+  * @param awsRegion AWS region
   * @param proxyConfiguration optional proxy configuration
   */
 class PanDomainAuthSettingsRefresher(
   val domain: String,
   val system: String,
+  bucketName: String,
   actorSystem: ActorSystem,
-  awsCredentialsProvider: AWSCredentialsProvider = new DefaultAWSCredentialsProviderChain(),
-  awsRegion: Option[Region] = Option(Region getRegion Regions.EU_WEST_1),
+  awsCredentialsProvider: AWSCredentialsProvider,
+  awsRegion: String,
```

Comment:
why the change in type?

Reply:
The latest AWS SDKs deprecated the previously code we used to construct the S3 client. Instead they suggest using the `withRegion` builder which either takes a string or a Regions object which I have found fiddly to build from the Region enums themselves.

It also means we don't need an SDK bump to target new AWS regions, although I'm not sure that's important.

Not a big deal though, we can keep it as the typed enum.

Reply:
The SDK advantage doesn't exist. When you pass in a string the SDK will find the Regions object for you so will hit the same problem. 

If you wanted to use a region not in the SDK you need to use withEndpoint instead of withRegion and pass in the endpoint URL.

I favour types here.

Reply:
:+1: for types! I think though that you might actually want to change the type like this:

```suggestion
  awsRegion: Option[com.amazonaws.regions.Regions],
```

...note that the type is now `Regions` rather than `Region` (and you get `Regions.EU_WEST_1` etc pre-constructed for you!) - and this object is accepted by the newer `com.amazonaws.client.builder.AwsClientBuilder#withRegion(com.amazonaws.regions.Regions)` builder method.

Reply:
Thanks everyone, makes sense. I've followed Roberto's suggestion but made the region mandatory. Any default region should be passed in from the calling application code.